openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/ConceptResource.java

```diff
@@ -96,10 +100,17 @@ protected Concept retrieveImpl(String uuid, RequestContext context, long snapsho
             for (FormField formField : chart.getFormFields()) {
                 Field field = formField.getField();
                 Concept fieldConcept = field.getConcept();
-                if (fieldConcept == null) continue;
-                ret.add(fieldConcept);
-                for (ConceptAnswer answer : fieldConcept.getAnswers(false)) {
-                    ret.add(answer.getAnswerConcept());
+                if (fieldConcept != null) {
+                    ret.add(fieldConcept);
+                    for (ConceptAnswer answer : fieldConcept.getAnswers(false)) {
+                        ret.add(answer.getAnswerConcept());
+                    }
+                }
+                // Fetch and add additional concepts that might have been stored in the field
+                // description.
+                for (Concept additionalConcept :
+                        getConceptsFromFieldDescription(field.getDescription())) {
+                    ret.add(additionalConcept);
```

Comment:
For (most) fields that are just a single concept, will it always have added the concept already on line 106 and try to add it again to the HashSet here?

If so, could lines 102-108 be removed?

Though maybe it's more intuitive as written, and then works in case we ever have a field that doesn't include its concept id in the description?